File: cpp/unittests/CourseScheduleTest.cpp
#include "CourseSchedule.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(CourseSchedule, Solution207_1) {
  const int NumCourses1 = 2;
  std::vector<std::pair<int, int>> Prerequisites1{{1, 0}};
  EXPECT_TRUE(Solution207_1().canFinish(NumCourses1, Prerequisites1));

  const int NumCourses2 = 2;
  std::vector<std::pair<int, int>> Prerequisites2{{1, 0}, {0, 1}};
  EXPECT_FALSE(Solution207_1().canFinish(NumCourses2, Prerequisites2));
}

File: java/src/test/java/io/binac/leetcode/CourseScheduleTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class CourseScheduleTest {
    private static final CourseSchedule.Solution1 SOLUTION1 = new CourseSchedule.Solution1();
    private static final CourseSchedule.Solution2 SOLUTION2 = new CourseSchedule.Solution2();

    private final int NUMCOURSES1 = 2;
    private final int[][] PREREQUISITES1 = {{1, 0}};
    private final boolean EXPECTED1 = true;

    private final int NUMCOURSES2 = 2;
    private final int[][] PREREQUISITES2 = {{1, 0}, {0, 1}};
    private final boolean EXPECTED2 = false;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.canFinish(NUMCOURSES1, PREREQUISITES1));
        assertEquals(EXPECTED2, SOLUTION1.canFinish(NUMCOURSES2, PREREQUISITES2));
    }

    @Test
    void test2() {
        assertEquals(EXPECTED1, SOLUTION2.canFinish(NUMCOURSES1, PREREQUISITES1));
        assertEquals(EXPECTED2, SOLUTION2.canFinish(NUMCOURSES2, PREREQUISITES2));
    }
}